### PR TITLE
Fix/issue 342

### PR DESCRIPTION
**Pull request by VivekFitkariwala**
This pull request has some major changes and it is not backward compatible.
### Changes
- Earlier we had assignments tables -> Individual Assignment and Group Assignment these tables are replaced with Individual Enrollment and Group Enrollment. From functionality new enrollment table will store the data of enrollment which means when the user mark the experiment point an enrollment document will be created based on the experiment and experiment user.
- Now the Enrollment table also stores the partition for which it got enrolled so in the future for implementing factorial/partial factorial it will be used.
- Indiviudla Enrollment table now also stores the workingGroupId for experiment with assignment type group. This will remove the Join cost of tables to calculate number of groups enrolled and number of users per group. This will also improve query from the ending criteria of per user per group.
- Queries are simplified because of the above design changes.
- Individual Enrollment also stores the enrollment codes.
- Exclusion table also stores the enrollment codes details [here](https://github.com/CarnegieLearningWeb/UpGrade/issues/27#issuecomment-1122076029)
- Earlier to calculate the enrollment details two tables Monitored Experiment Point and Assignment Table need to be joined to get the data. Now with the above change of Enrollment Table only single table will be queried to get those details so it will improve the database performance overall.
### Things which are left
- Export data part is commented for now because the format is going to get changed so I will be implemented at that time.
- Exclusion code `PARTICIPANT_ON_EXCLUSION_LIST` and `GROUP_ON_EXCLUSION_LIST` is left to be added as this will get affected by segment implementation.

### CODE REVIEW

File: backend/packages/Upgrade/.env.test [hunk 1]
```diff
@@ -19,7 +19,7 @@ LOG_OUTPUT=dev
 # PostgreSQL DATABASE
 #
 TYPEORM_CONNECTION=postgres
-TYPEORM_HOST=postgres
+TYPEORM_HOST=localhost
```
Comment (ppratikcr7) [inline backend/packages/Upgrade/.env.test:22]: Can you revert this back to 'postgres'. This is just required for local testing.

**Comment by danoswaltCL** [comment]
i will try and pull this down and spend some time looking into this today in local docker. before this gets merged, since it is a major breaking change, I think it would make sense to bump base version to 2.0?

**Review by danoswaltCL — DISMISSED** [review]
This looks good to me, I've pulled this down and done some of the locust load tests / manual testing like how @ppratikcr7 demo'd. I'm eager to get this code in and tested in QA, will be easier to spot edge cases. thx @VivekFitkariwala for suggesting and implementing this change, I think it will make a big difference.

**Review by ppratikcr7 — DISMISSED** [review]

**Comment by ppratikcr7** [comment]
Looks good to me as well.

**Review by danoswaltCL — DISMISSED** [review]

**Review by ppratikcr7 — APPROVED** [review]

**Review by jreddig — APPROVED** [review]
Checked this out locally and ran manual/locust tests. Looks good!

**Comment by ppratikcr7** [comment]
> Checked this out locally and ran manual/locust tests. Looks good!

@jreddig Great. We want to merge this PR but would need a clean db for this. Can you do that for dev env and let us know.

**Comment by jreddig** [comment]
> @jreddig Great. We want to merge this PR but would need a clean db for this. Can you do that for dev env and let us know.

dev db cleared and ready for merge
